databroker/src/grpc/kuksa_val_v1/val.rs

```diff
@@ -344,12 +283,146 @@ impl proto::val_server::Val for broker::DataBroker {
+                                                        let message = format!("{} not found", entry.path);
+                                                        errors.push(proto::DataEntryError {
+                                                            path: entry.path.clone(),
+                                                            error: Some(proto::Error {
+                                                                code: 404,
+                                                                reason: "not_found".to_string(),
+                                                                message,
+                                                            })
+                                                        })
+                                                    }
+                                                },
+                                                None => {
+                                                    errors.push(proto::DataEntryError {
+                                                        path: "".to_string(),
+                                                        error: Some(proto::Error {
+                                                            code: 400,
+                                                            reason: "invalid_data".to_string(),
+                                                            message: "Data present in the request is invalid: Path is required".to_string()
+                                                        })
+                                                    })
+                                                }
+                                            }
+                                        }
+
+                                        match broker.update_entries(updates).await {
+                                            Ok(_) => {}
+                                            Err(err) => {
+                                                debug!("Failed to set datapoint: {:?}", err);
+                                                for (id, error) in err.into_iter() {
+                                                    if let Some(metadata) = broker.get_metadata(id).await {
+                                                        let path = metadata.path.clone();
+                                                        let data_entry_error = convert_to_data_entry_error(&path, &error);
+                                                        errors.push(data_entry_error);
+                                                    }
+                                                }
+
+                                                if let Err(err) = sender.send(
+                                                    Ok(proto::StreamedUpdateResponse {
+                                                        errors,
+                                                        error: None, // TODO take first element of errors
```

Comment:
I absolutely could not get this to work, once I got the first element from errors, errors was unhappy due to it being a move :(

Reply:
what you would do is do an 'errors.clone()' then you will not move the original errors array.